update_factorio.py

```diff
@@ -138,11 +138,28 @@ def verbose_aware_exec(exec_args, verbose=False):
         raise
 
 
+def get_username_token(factoriopath):
+    with open(os.path.normpath(factoriopath + '/../../../player-data.json')) as data_file:  
+        data = json.load(data_file)
+    glob['user'] = data["service-username"]
+    glob['token'] = data["service-token"]   
+
 def main():
     args = parser.parse_args()
     glob['verbose'] = args.verbose
 
-    j = get_updater_data(args.user, args.token)
+    if args.user and args.token:
+        glob['user'] = args.user
+        glob['token'] = args.token
+    else:
+        if args.apply_to:
+            get_username_token(args.apply_to)
+            print("No username or token provided, reading from player-data.json")
+        else:
+            print("Username/Token not available")
+            exit()
```

Comment:
Headless server doesn't need user+token to be updated. We don't need to always exit.